addons/udes_security/models/ir_attachment.py

```diff
@@ -136,6 +165,22 @@ def write(self, vals):
         if "active" in vals and not self.env.context.get("skip_active_check"):
             # Prevent user from manually setting attachment to active/inactive
             del vals["active"]
+
+        if vals.get("url", False):
+            # Restrict URLs to relative paths (starting with "/") or HTTPS (starting with "https://")
+            # This ensures only safe URL types are allowed, preventing schemes like "file://" or "sftp://"
+            # from bypassing security checks.
+            if not (vals["url"].startswith("/") or vals["url"].startswith("https://")):
+                # Raise an error if URL is not a valid type
+                raise UserError("Invalid URL: Only secure HTTPS links or relative URLs are allowed. Please update the URL format and try again.")
+            if "mimetype" in vals and vals["mimetype"] in allowed_image_mimetypes:
```

Comment:
If you create a url attachment and set it to something like `https://google.com` then write it and change it to `https://some_site.com/image.png` then this code doesn't get run. The mimetype is left as `application/octet-stream` on `self` and this code doesn't get run, thus the image is not sanitised.

Reply:
Addressed in c5b227f9c550cb52c1e88946fbac57a872fe716d